[BUG] Virtual Node broken when radio restarts unexpectedly

**Describe the bug**
I was having issues where the MQTT proxy would rapidly keep reconnecting to the virtual node.  meshmonitor would report it sent the config bundle, but then the proxy would report connection lost and this would repeat several times a second.

After researching it and testing, the meshtastic CLI does the same thing when in the "broken" situation, so I think it's related to the virtual node.

Right before things go poorly, the radio has restarted causing the config to be reloaded.  During that time, the proxy tries to connect and get information which is not available yet:

```
Jan 14 04:55:10 bento meshmonitor[2525412]: [INFO] Virtual node: Using hybrid approach - rebuilding dynamic data from database
Jan 14 04:55:10 bento meshmonitor[2525412]: [WARN] ⚠️ Init config capture not yet complete, returning partial cache
Jan 14 04:55:10 bento meshmonitor[2525412]: [WARN] Virtual node: No local node info available, skipping MyNodeInfo - clients may have incorrect identity!
```

Then the errors start:

```
Jan 14 04:55:10 bento meshmonitor-mqtt-proxy[2525470]: 2026-01-14 10:55:10,949 [WARNING] mqtt-proxy: Meshtastic connection reported LOST!
Jan 14 04:55:10 bento meshmonitor[2525412]: [ERROR] Virtual node: Failed to send to vn-1: write EPIPE
Jan 14 04:55:10 bento meshmonitor[2525412]: [ERROR] Virtual node client vn-1 error: write EPIPE
Jan 14 04:55:10 bento meshmonitor[2525412]: [INFO] 📱 Virtual node client disconnected: vn-1 (0 remaining)
Jan 14 04:55:10 bento meshmonitor[2525412]: [ERROR] Virtual node: Error sending initial config to vn-1: Error: write EPIPE
Jan 14 04:55:10 bento meshmonitor[2525412]:     at afterWriteDispatched (node:internal/stream_base_commons:159:15)
Jan 14 04:55:10 bento meshmonitor[2525412]:     at writeGeneric (node:internal/stream_base_commons:150:3)
Jan 14 04:55:10 bento meshmonitor[2525412]:     at Socket._writeGeneric (node:net:966:11)
Jan 14 04:55:10 bento meshmonitor[2525412]:     at Socket._write (node:net:978:8)
Jan 14 04:55:10 bento meshmonitor[2525412]:     at writeOrBuffer (node:internal/streams/writable:570:12)
Jan 14 04:55:10 bento meshmonitor[2525412]:     at _write (node:internal/streams/writable:499:10)
Jan 14 04:55:10 bento meshmonitor[2525412]:     at Writable.write (node:internal/streams/writable:508:10)
Jan 14 04:55:10 bento meshmonitor[2525412]:     at file:///app/dist/server/virtualNodeServer.js:633:27
Jan 14 04:55:10 bento meshmonitor[2525412]:     at new Promise (<anonymous>)
Jan 14 04:55:10 bento meshmonitor[2525412]:     at VirtualNodeServer.sendToClient (file:///app/dist/server/virtualNodeServer.js:632:16) {
Jan 14 04:55:10 bento meshmonitor[2525412]:   errno: -32,
Jan 14 04:55:10 bento meshmonitor[2525412]:   code: 'EPIPE',
Jan 14 04:55:10 bento meshmonitor[2525412]:   syscall: 'write'
Jan 14 04:55:10 bento meshmonitor[2525412]: }
Jan 14 04:55:12 bento meshmonitor[2525412]: [INFO] 📡 Requesting LoRa config from device...
```

After this, the virtual node appears to be in a stuck state, even though the radio and meshmonitor itself are working:
```
Jan 14 04:56:11 bento meshmonitor-mqtt-proxy[2525470]: 2026-01-14 10:56:11,228 [ERROR] mqtt-proxy: Connection LOST for 60s. Forcing restart...
Jan 14 04:56:11 bento meshmonitor-mqtt-proxy[2530107]: 2026-01-14 10:56:11,611 [INFO] mqtt-proxy: MQTT Proxy starting (interface: TCP)...
Jan 14 04:56:11 bento meshmonitor-mqtt-proxy[2530107]: 2026-01-14 10:56:11,612 [INFO] mqtt-proxy: Creating TCP interface...
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] 📱 Virtual node client connected: vn-2 (1 total)
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Received 6 bytes from vn-2
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Parsed message from vn-2: {
Jan 14 04:56:11 bento meshmonitor[2525412]:   "wantConfigId": 1513156450
Jan 14 04:56:11 bento meshmonitor[2525412]: }
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Client vn-2 requesting config with ID 1513156450
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Starting to send initial config to vn-2 (ID: 1513156450)
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Using hybrid approach - rebuilding dynamic data from database
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Sending MyNodeInfo with nodeNum=3680020484 (!db58a804) to vn-2
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ Sent 78 fresh NodeInfo entries from database
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ Sent 39 cached static messages (config, channels, metadata)
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Sending ConfigComplete to vn-2 with ID 1513156450...
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ ConfigComplete sent to vn-2 (ID: 1513156450)
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: ✅ Initial config fully sent to vn-2 (119 total messages - 78 fresh NodeInfo + 39 cached static)
Jan 14 04:56:11 bento meshmonitor-mqtt-proxy[2530107]: 2026-01-14 10:56:11,737 [WARNING] mqtt-proxy: Meshtastic connection reported LOST!
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Received 6 bytes from vn-2
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Parsed message from vn-2: {
Jan 14 04:56:11 bento meshmonitor[2525412]:   "wantConfigId": 2322632392
Jan 14 04:56:11 bento meshmonitor[2525412]: }
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Client vn-2 requesting config with ID 2322632392
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Starting to send initial config to vn-2 (ID: 2322632392)
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Using hybrid approach - rebuilding dynamic data from database
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Sending MyNodeInfo with nodeNum=3680020484 (!db58a804) to vn-2
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ Sent 78 fresh NodeInfo entries from database
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ Sent 39 cached static messages (config, channels, metadata)
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: Sending ConfigComplete to vn-2 with ID 2322632392...
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ ConfigComplete sent to vn-2 (ID: 2322632392)
Jan 14 04:56:11 bento meshmonitor[2525412]: [INFO] Virtual node: ✅ Initial config fully sent to vn-2 (119 total messages - 78 fresh NodeInfo + 39 cached static)
Jan 14 04:56:11 bento meshmonitor-mqtt-proxy[2530107]: 2026-01-14 10:56:11,761 [WARNING] mqtt-proxy: Meshtastic connection reported LOST!
```

trying with the meshtastic CLI shows the same behavior:
```
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Received 5 bytes from vn-10
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Parsed message from vn-10: {
Jan 14 06:36:13 bento meshmonitor[2525412]:   "wantConfigId": 192424272
Jan 14 06:36:13 bento meshmonitor[2525412]: }
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Client vn-10 requesting config with ID 192424272
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Starting to send initial config to vn-10 (ID: 192424272)
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Using hybrid approach - rebuilding dynamic data from database
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Sending MyNodeInfo with nodeNum=3680020484 (!db58a804) to vn-10
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ Sent 77 fresh NodeInfo entries from database
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ Sent 39 cached static messages (config, channels, metadata)
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Sending ConfigComplete to vn-10 with ID 192424272...
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ ConfigComplete sent to vn-10 (ID: 192424272)
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: ✅ Initial config fully sent to vn-10 (118 total messages - 77 fresh NodeInfo + 39 cached static)
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Received 6 bytes from vn-10
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Parsed message from vn-10: {
Jan 14 06:36:13 bento meshmonitor[2525412]:   "wantConfigId": 2288570686
Jan 14 06:36:13 bento meshmonitor[2525412]: }
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Client vn-10 requesting config with ID 2288570686
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Starting to send initial config to vn-10 (ID: 2288570686)
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Using hybrid approach - rebuilding dynamic data from database
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Sending MyNodeInfo with nodeNum=3680020484 (!db58a804) to vn-10
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ Sent 77 fresh NodeInfo entries from database
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ Sent 39 cached static messages (config, channels, metadata)
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Sending ConfigComplete to vn-10 with ID 2288570686...
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: ✓ ConfigComplete sent to vn-10 (ID: 2288570686)
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: ✅ Initial config fully sent to vn-10 (118 total messages - 77 fresh NodeInfo + 39 cached static)
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Received 6 bytes from vn-10
Jan 14 06:36:13 bento meshmonitor[2525412]: [INFO] Virtual node: Parsed message from vn-10: {
Jan 14 06:36:13 bento meshmonitor[2525412]:   "wantConfigId": 310931657
```

**To Reproduce**
Steps to reproduce the behavior:
1. Have the MQTT proxy enabled against the virtual node
1. The radio restarts unexpectedly
1. While the real radio is restarting and config is loading into meshmonitor, have the proxy connect to the virtual node

**Expected behavior**
The virtual node should either not accept connections while re-establishing, like it does when first starting up, or it should at least detect the broken state and reconfigure itself once the real radio is up.

**Environment (please complete the following information):**
* Host OS: Linux, AlmaLinux 9.7
* Deployment type: Podman (Docker)
  * if Docker, which Container version are you using?
    * ghcr.io/yeraze/meshmonitor:latest
    * ghcr.io/ln4cy/mqtt-proxy:master

--- a/src/server/virtualNodeServer.ts
+++ b/src/server/virtualNodeServer.ts
@@ -562,6 +562,14 @@ export class VirtualNodeServer extends EventEmitter {
   private async sendInitialConfig(clientId: string, configId?: number): Promise<void> {
     logger.info(`Virtual node: Starting to send initial config to ${clientId}${configId ? ` (ID: ${configId})` : ''}`);
     try {
+      // Check if config capture is complete before sending anything
+      // This prevents sending partial/incomplete config when the radio is restarting
+      if (!this.config.meshtasticManager.isInitConfigCaptureComplete()) {
+        logger.warn(`Virtual node: Config capture not yet complete, cannot send config to ${clientId}`);
+        logger.warn(`Virtual node: Physical node may be restarting - client should retry after initialization completes`);
+        return;
+      }
+
       // Get cached init config with type metadata from meshtasticManager
       const cachedMessages = this.config.meshtasticManager.getCachedInitConfig();
 

